### PR TITLE
VIITE-750 - Visualisation of project screen needs addition

### PR DESCRIPTION
**Pull request by pedrosag**
*Added layer to draw the reserved (NotHandled) roads from a project
*Added the projectStyler in order to attribute the correct style to the reserved features

### CODE REVIEW

File: viite-UI/src/model/RoadCollection.js [hunk 1]
```diff
@@ -294,5 +294,26 @@
     var roadIsUnknown = function(road){
       return  0 === road.roadNumber && 1 === road.anomaly && 0 === road.roadLinkType && 0 === road.roadPartNumber && 99 === road.trackCode;
     };
+    
+    this.findReservedProjectLinks = function(boundingBox, zoomLevel, projectId) {
+      backend.getProjectLinks({boundingBox: boundingBox, zoom: zoomLevel, projectId: projectId}, function(fetchedLinks) {
+        var notHandledLinks = _.chain(fetchedLinks).flatten().filter(function (link) {
+          return link.status === 0;
```
Comment (oappicgi) [inline viite-UI/src/model/RoadCollection.js]: we could have unchanged  variable instead of 0

File: viite-UI/src/view/LinkPropertyLayer.js [hunk 2]
```diff
@@ -1056,12 +1067,21 @@
...
       setGeneralOpacity(0.2);
-      var projectFeature = _.filter(roadLayer.layer.getSource().getFeatures(), function (ft) {
-        return ft.roadLinkData.linkId === featureLinkId;
+        var boundingBox = map.getView().calculateExtent(map.getSize());
+        var zoomLevel = map.getView().getZoom();
+        roadCollection.findReservedProjectLinks(boundingBox, zoomLevel, featureLinkId);
+    });
+
+    eventbus.on('linkProperties:highlightReservedRoads', function(reservedOL3Features){
+    //              var style = projectLinkStyler.getProjectLinkStyle().getStyle( feature.projectLinkData, {zoomLevel: map.getView().getZoom()});
```
Comment (oappicgi) [inline viite-UI/src/view/LinkPropertyLayer.js]: do we need this commented out code?

**Comment by tvenhola** [comment]
:+1:

**Comment by pedrosag** [comment]
@oappicgi regarding the comment I had no trouble either creating new links from black roads on my end and the form is supporting standard operations to the ReservedLinks

image

**Pull request merged**